src/puptoo/process/profile.py

```diff
@@ -735,6 +740,21 @@ def system_profile(
         if subscription_manager_facts.get('conversions.activity') == 'conversion':
             profile["conversions"]["activity"] = True
 
+    # profile["third_party_services"]:
+    #   containing information about system facts of third party services
+    third_party_services = {}
+
+    crowdstrike_facts = {}
+    if falconctl_aid:
+        crowdstrike_facts["falcon_aid"] = falconctl_aid.aid
+    if falconctl_backend:
+        crowdstrike_facts["falcon_backend"] = falconctl_backend.backend
+    if crowdstrike_facts:
+        third_party_services["crowdstrike"] = crowdstrike_facts
+
+    if third_party_services:
+        profile["third_party_services"] = third_party_services
```

Comment:
The `third_party_services` variable seems not required, how about making it simple as bellow?
```python
    crowdstrike_facts = {}
    if falconctl_aid:
        crowdstrike_facts["falcon_aid"] = falconctl_aid.aid
    if falconctl_backend:
        crowdstrike_facts["falcon_backend"] = falconctl_backend.backend
    if crowdstrike_facts:
        profile["third_party_services"] = dict(crowdstrike=crowdstrike_facts)
```

Reply:
@xiangce Yep , the `third_party_services` variable is not required, and your version of code also works perfect logically. 

As in the future,  we would have other attributes be added into  `profile["third_party_services"]`, so I put the `third_party_services` out of the `if crowdstrike_facts` check to have a better compatibility for later update, for example, we can add new attribute right before the `if third_party_services:` line w/o touching the current code.

Reply:
Sure, good to me